From a5b0d115f35eca8712ba23dc277842a77da99580 Mon Sep 17 00:00:00 2001
From: Jiawei Huang <jiawei@tigera.io>
Date: Sun, 14 Jan 2024 12:19:01 -0800
Subject: [PATCH] Update dependencies to reduce CVEs

This changeset updates Elasticseach dependencies for v7.17.16 to reduce
CVEs detected. The ML artifacts change in [1] is reverted to v7.17.6
behavior as downloading artifacts from `artifacts-snapshot.elastic.co`
doesn't seem to work well with custom Elasticseach builds from source.

[1] https://github.com/elastic/elasticsearch/pull/89985
---
 build-tools-internal/version.properties |   6 +-
 gradle/verification-metadata.xml        | 107 +++++++++++++++++++++++-
 plugins/repository-gcs/build.gradle     |   4 +-
 x-pack/plugin/ml/build.gradle           |  23 +----
 x-pack/plugin/security/build.gradle     |   4 +-
 x-pack/plugin/vector-tile/build.gradle  |   4 +-
 x-pack/snapshot-tool/build.gradle       |   4 +-
 7 files changed, 120 insertions(+), 32 deletions(-)

diff --git a/build-tools-internal/version.properties b/build-tools-internal/version.properties
index cb50a611205..a8004b69cc1 100644
--- a/build-tools-internal/version.properties
+++ b/build-tools-internal/version.properties
@@ -8,7 +8,7 @@ spatial4j         = 0.7
 jts               = 1.15.0
 jackson           = 2.14.2
 snakeyaml         = 1.33
-icu4j             = 62.1
+icu4j             = 68.2
 supercsv          = 2.4.0
 # when updating log4j, please update also docs/java-api/index.asciidoc
 log4j             = 2.17.1
@@ -16,7 +16,7 @@ slf4j             = 1.6.2
 
 jna               = 5.10.0
 
-netty             = 4.1.94.Final
+netty             = 4.1.100.Final
 joda              = 2.10.10
 
 commons_lang3                   = 3.9
@@ -25,7 +25,7 @@ commons_lang3                   = 3.9
 #  - plugins/ingest-attachment (transitive dependency, check the upstream POM)
 #  - distribution/tools/plugin-cli
 #  - x-pack/plugin/security
-bouncycastle=1.64
+bouncycastle=1.70
 # test dependencies
 randomizedrunner  = 2.8.0
 junit             = 4.12
diff --git a/gradle/verification-metadata.xml b/gradle/verification-metadata.xml
index 62e1364e512..987db61baeb 100644
--- a/gradle/verification-metadata.xml
+++ b/gradle/verification-metadata.xml
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<verification-metadata xmlns="https://schema.gradle.org/dependency-verification" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://schema.gradle.org/dependency-verification https://schema.gradle.org/dependency-verification/dependency-verification-1.2.xsd">
+<verification-metadata xmlns="https://schema.gradle.org/dependency-verification" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://schema.gradle.org/dependency-verification https://schema.gradle.org/dependency-verification/dependency-verification-1.3.xsd">
    <configuration>
       <verify-metadata>false</verify-metadata>
       <verify-signatures>false</verify-signatures>
@@ -809,6 +809,11 @@
             <sha256 value="a204ec68748a7b26351ae37a311e8de468f248d1916d5f8dbe812c1289d0a0ff" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.google.protobuf" name="protobuf-java" version="3.21.9">
+         <artifact name="protobuf-java-3.21.9.jar">
+            <sha256 value="1b78b4a76a71512debfdff8f8fc5aef6bfd459f65758fecf7aff245e6e6301e4" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.google.protobuf" name="protobuf-java" version="3.4.0">
          <artifact name="protobuf-java-3.4.0.jar">
             <sha256 value="dce7e66b32456a1b1198da0caff3a8acb71548658391e798c79369241e6490a4" origin="Generated by Gradle"/>
@@ -819,6 +824,11 @@
             <sha256 value="bf320ed076000e1d8c7cbf7601b056acaecab80f75b9a659b9f6398d0d7e3f79" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.google.protobuf" name="protobuf-java-util" version="3.21.9">
+         <artifact name="protobuf-java-util-3.21.9.jar">
+            <sha256 value="a78226d02551ca2cb34b24b4168a75b4a84041cc3333f3725bb5ad9806a80b77" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.google.re2j" name="re2j" version="1.1">
          <artifact name="re2j-1.1.jar">
             <sha256 value="24ada84d1b5de584e3e84b06f0c7dd562cee6eafe8dea8083bd8eb123823bbe7" origin="Generated by Gradle"/>
@@ -864,6 +874,11 @@
             <sha256 value="e5b0962e566003d68bf3ec7d87b60e279f5cc7aa93cbc4056432803046512478" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="com.ibm.icu" name="icu4j" version="68.2">
+         <artifact name="icu4j-68.2.jar">
+            <sha256 value="9bd7bf869a44ba8aeb0cddd7e6616e88cd4795ba5bfce2230447cb0e185a646c" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="com.jamesmurty.utils" name="java-xmlbuilder" version="0.4">
          <artifact name="java-xmlbuilder-0.4.jar">
             <sha256 value="681e53c4ffd59fa12068803b259e3a83d43f07a47c112e748a187dee179eb31f" origin="Generated by Gradle"/>
@@ -1394,6 +1409,11 @@
             <sha256 value="a3ebec96768ee4a2d3db44597e84cea2d0bdd68ca04822397980ea9f67075a86" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-buffer" version="4.1.100.Final">
+         <artifact name="netty-buffer-4.1.100.Final.jar">
+            <sha256 value="462874b44ee782fbefec64078cda6eb8e7bf9f0e0af71a928ef4c1f2d564f7ee" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-buffer" version="4.1.42.Final">
          <artifact name="netty-buffer-4.1.42.Final.jar">
             <sha256 value="7b0171a4e8bcd573e08d9f2bba053c67b557ab5012106a5982ccbae5743814c0" origin="Generated by Gradle"/>
@@ -1409,6 +1429,11 @@
             <sha256 value="8066ee7c49f9f29da96ee62f7cb13bee022cb4b68e51437b33da3b6d01398f13" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-codec" version="4.1.100.Final">
+         <artifact name="netty-codec-4.1.100.Final.jar">
+            <sha256 value="180a01ed67af399602e24ff1c32864e7f57f57c4a0fa5e9ab3fe9b0e5e9cf051" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-codec" version="4.1.42.Final">
          <artifact name="netty-codec-4.1.42.Final.jar">
             <sha256 value="e96ced697fb7df589da7c20c995e01f75a9cb246be242bbc4cd3b4af424ff189" origin="Generated by Gradle"/>
@@ -1424,6 +1449,11 @@
             <sha256 value="91243776ad68b4d8e39eafb9ec115e1b8fa9aecd147b12ef15bb691639498328" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-codec-dns" version="4.1.100.Final">
+         <artifact name="netty-codec-dns-4.1.100.Final.jar">
+            <sha256 value="f7f55bee95d2467e8bda6dda980a2830b351b0dd6f0f87fe5114645035ee55ca" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-codec-dns" version="4.1.53.Final">
          <artifact name="netty-codec-dns-4.1.53.Final.jar">
             <sha256 value="747098d17592d092afdfd8c746e099c3b739986461ebcfa68d782f4b5b19b8ae" origin="Generated by Gradle"/>
@@ -1434,6 +1464,11 @@
             <sha256 value="b345048b7692204803b49eb11f5203b52e18aa7647f8b77dd63118fd8d5fd2a2" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-codec-http" version="4.1.100.Final">
+         <artifact name="netty-codec-http-4.1.100.Final.jar">
+            <sha256 value="326811d249cb0e5555e78e026e877834e792261c38f0666d80464426695d9590" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-codec-http" version="4.1.53.Final">
          <artifact name="netty-codec-http-4.1.53.Final.jar">
             <sha256 value="48a81eec058871c4eaf66670f824a76f403579c53ac7a6840dd6ace9dfc4ad4e" origin="Generated by Gradle"/>
@@ -1444,6 +1479,11 @@
             <sha256 value="1ada4580f68cd17a534fb3c0337087073223a76cb77304dbe5a1b19df3d53c2f" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-codec-http2" version="4.1.100.Final">
+         <artifact name="netty-codec-http2-4.1.100.Final.jar">
+            <sha256 value="23b4a74350f4cf8d41b93fb93d52b5050667d8d53fffc385672c86eab83b8749" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-codec-http2" version="4.1.53.Final">
          <artifact name="netty-codec-http2-4.1.53.Final.jar">
             <sha256 value="6c777f308be5bb53f33925e695dc97f0b88688e2f1937c2c7a3975e977362df4" origin="Generated by Gradle"/>
@@ -1454,6 +1494,11 @@
             <sha256 value="8fbd2e95abec6155b60ed3c9c1600ed4e17ffe3f053cd5a40677d879c0af961f" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-codec-socks" version="4.1.100.Final">
+         <artifact name="netty-codec-socks-4.1.100.Final.jar">
+            <sha256 value="608a453b90f8384ba4efcdc6db7f899a1f10b9ea1890954696e6cfac45ff1ba9" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-codec-socks" version="4.1.53.Final">
          <artifact name="netty-codec-socks-4.1.53.Final.jar">
             <sha256 value="8de51d13f48da8eece7df43aa00d6d2923a215668526d8888879fa1534e9aafe" origin="Generated by Gradle"/>
@@ -1464,6 +1509,11 @@
             <sha256 value="1c0b4ee0b623402dd5bb78fd4a69bf808da2510524f489a24b0f3cc58ded046e" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-common" version="4.1.100.Final">
+         <artifact name="netty-common-4.1.100.Final.jar">
+            <sha256 value="d2908301f1ac6f2910900742473c15d701765d3d4467acdb1eebb9df3aa82885" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-common" version="4.1.42.Final">
          <artifact name="netty-common-4.1.42.Final.jar">
             <sha256 value="3d0a918d78292eeca02a7bb2188daa4e5053b6e29b71e6308309033e121242b5" origin="Generated by Gradle"/>
@@ -1479,6 +1529,11 @@
             <sha256 value="cb8d84a3e63aea90d0d7a333a02e50ac751d2b05db55745d981b5eff893f647b" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-handler" version="4.1.100.Final">
+         <artifact name="netty-handler-4.1.100.Final.jar">
+            <sha256 value="0e10e584c2e7fdf7f4804e14760ed987003f1b62ab982f62eaf13a9892793d3a" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-handler" version="4.1.42.Final">
          <artifact name="netty-handler-4.1.42.Final.jar">
             <sha256 value="11eda86500c33b9d386719b5419f513fd9c097d13894f25dd0c75b610d636e03" origin="Generated by Gradle"/>
@@ -1494,6 +1549,11 @@
             <sha256 value="8e50719a9ab89e33ef85c5f36d780e0d7056b3f768b07d261d87baed7094eb3c" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-handler-proxy" version="4.1.100.Final">
+         <artifact name="netty-handler-proxy-4.1.100.Final.jar">
+            <sha256 value="686dbc2e61407f216d6cb267dd7954896f851dd34b58be3e757c5a89f20a5e67" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-handler-proxy" version="4.1.53.Final">
          <artifact name="netty-handler-proxy-4.1.53.Final.jar">
             <sha256 value="1f4b5a540f596e45bcde7c53f4c02780aa5e56eea4103eee78b7d104f90c97b1" origin="Generated by Gradle"/>
@@ -1504,6 +1564,11 @@
             <sha256 value="378738bfeb84c81228ae5de209f964ca87849f9a720c916c77f7cba6494bb1ec" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-resolver" version="4.1.100.Final">
+         <artifact name="netty-resolver-4.1.100.Final.jar">
+            <sha256 value="c42c481c776e9d367a45cc3a67a06f65897d280334eb30b2362b8c55b7523f4f" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-resolver" version="4.1.42.Final">
          <artifact name="netty-resolver-4.1.42.Final.jar">
             <sha256 value="89768242b6b7cce9bd9f5945ad21d1b4bae515c6b1bf03a8af5d1899779cebc9" origin="Generated by Gradle"/>
@@ -1519,6 +1584,11 @@
             <sha256 value="bd26e9bc5e94e2d3974a93fdf921658eff4f033bfd4c5208607760ab54298617" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-resolver-dns" version="4.1.100.Final">
+         <artifact name="netty-resolver-dns-4.1.100.Final.jar">
+            <sha256 value="4bafbd3e571ed2d9fb3f9550f8348ba37c183eda4a253fba5354166e409624c4" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-resolver-dns" version="4.1.53.Final">
          <artifact name="netty-resolver-dns-4.1.53.Final.jar">
             <sha256 value="bb63961902ce4a4724e80efde16a63b06f871a2cb2387eb90aa2c3663f17db90" origin="Generated by Gradle"/>
@@ -1529,6 +1599,11 @@
             <sha256 value="d6a0871adbc47ce4b08ab7c23c6e32047bd197641f5f3fd35d165faed79a89e8" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-transport" version="4.1.100.Final">
+         <artifact name="netty-transport-4.1.100.Final.jar">
+            <sha256 value="b1deeceedab3734cdb959c55f4be5ab4a667a8aed59121ff93763f49470f5470" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-transport" version="4.1.42.Final">
          <artifact name="netty-transport-4.1.42.Final.jar">
             <sha256 value="dfa817a156ea263aa9ad8364a2e226527665c9722aca40a7945f228c2c14f1da" origin="Generated by Gradle"/>
@@ -1554,6 +1629,11 @@
             <sha256 value="e0685ad59b8c00eff7abe0caed8db08b9a0471f2c848958515008bc48fe73238" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="io.netty" name="netty-transport-native-unix-common" version="4.1.100.Final">
+         <artifact name="netty-transport-native-unix-common-4.1.100.Final.jar">
+            <sha256 value="5d888230a04c4a3e647c64e933cefb64fd49056f969bfb734c8a3fcedf0bea8a" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="io.netty" name="netty-transport-native-unix-common" version="4.1.42.Final">
          <artifact name="netty-transport-native-unix-common-4.1.42.Final.jar">
             <sha256 value="508fba9128da78bd775ba854d71917ceb2b00b95a7600254f54a277a06761a86" origin="Generated by Gradle"/>
@@ -1837,6 +1917,11 @@
             <sha256 value="0972bbc99437c4163acd09b630e6c77eab4cfab8a9594621c95466c0c6645396" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="net.minidev" name="accessors-smart" version="2.4.8">
+         <artifact name="accessors-smart-2.4.8.jar">
+            <sha256 value="7dd705aa1ac0e030f8ee2624e8e77239ae1eef6ccc2621c0b8c189866ee1c42c" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="net.minidev" name="json-smart" version="2.3">
          <artifact name="json-smart-2.3.jar">
             <sha256 value="903f48c8aa4c3f6426440b8d32de89fa1dc23b1169abde25e4e1d068aa67708b" origin="Generated by Gradle"/>
@@ -3102,6 +3187,11 @@
             <sha256 value="2e2ec8fe0cf873979f630ae4d35e7ede3390321279b7a15de9deed3f3430990c" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="org.apache.santuario" name="xmlsec" version="2.1.8">
+         <artifact name="xmlsec-2.1.8.jar">
+            <sha256 value="75b6f42147bc8e8718a764821439937fee224a8322ac43ff0b2b406edd7f5c2b" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.apache.servicemix.bundles" name="org.apache.servicemix.bundles.antlr" version="2.7.7_5">
          <artifact name="org.apache.servicemix.bundles.antlr-2.7.7_5.jar">
             <sha256 value="3902794d36d9b81da1b7e697f21ed04ccae276cc116eecc640a4cd0fff2691f2" origin="Generated by Gradle"/>
@@ -3165,6 +3255,11 @@
             <sha256 value="9b88abe9e981d17113a081a4b4261f4082ccdf9144a8a8373fc460c586f6b13a" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="org.bouncycastle" name="bcmail-jdk15on" version="1.70">
+         <artifact name="bcmail-jdk15on-1.70.jar">
+            <sha256 value="ff6cde372bcabca182e40c1cc5d9b1f9eb73370cad286ce362d3747aff15f230" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.bouncycastle" name="bcpg-fips" version="1.0.4">
          <artifact name="bcpg-fips-1.0.4.jar">
             <sha256 value="b73c80be1099c4756c088cb457a82040509b787519af5c72c9c3d1bff357565e" origin="Generated by Gradle"/>
@@ -3185,6 +3280,11 @@
             <sha256 value="84669138b1d99143e2c009024f67824ab8d3edb9b05b7591f5ebfb020a4bda71" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="org.bouncycastle" name="bcpkix-jdk15on" version="1.70">
+         <artifact name="bcpkix-jdk15on-1.70.jar">
+            <sha256 value="e5b9cb821df57f70b0593358e89c0e8d7266515da9d088af6c646f63d433c07c" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.bouncycastle" name="bcprov-jdk15on" version="1.60">
          <artifact name="bcprov-jdk15on-1.60.jar">
             <sha256 value="7f1a0e6badab38666f8467a9a0ee96656b2f8ec8623867ed34f3cdc173b7ee07" origin="Generated by Gradle"/>
@@ -3200,6 +3300,11 @@
             <sha256 value="e469bd39f936999f256002631003ff022a22951da9d5bd9789c7abfa9763a292" origin="Generated by Gradle"/>
          </artifact>
       </component>
+      <component group="org.bouncycastle" name="bcprov-jdk15on" version="1.70">
+         <artifact name="bcprov-jdk15on-1.70.jar">
+            <sha256 value="8f3c20e3e2d565d26f33e8d4857a37d0d7f8ac39b62a7026496fcab1bdac30d4" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
       <component group="org.bouncycastle" name="bctls-fips" version="1.0.9">
          <artifact name="bctls-fips-1.0.9.jar">
             <sha256 value="ada8549467dda0017e098f2362bd7f00841244351d1fc6f89b9da5ff25890151" origin="Generated by Gradle"/>
diff --git a/plugins/repository-gcs/build.gradle b/plugins/repository-gcs/build.gradle
index 8401bb8a1c3..638669b2f92 100644
--- a/plugins/repository-gcs/build.gradle
+++ b/plugins/repository-gcs/build.gradle
@@ -37,8 +37,8 @@ dependencies {
   api 'com.google.api:api-common:2.3.1'
   api 'com.google.api:gax:2.20.1'
   api 'org.threeten:threetenbp:1.6.5'
-  api 'com.google.protobuf:protobuf-java-util:3.17.3'
-  api 'com.google.protobuf:protobuf-java:3.21.7'
+  api 'com.google.protobuf:protobuf-java-util:3.21.9'
+  api 'com.google.protobuf:protobuf-java:3.21.9'
   api 'com.google.code.gson:gson:2.10'
   api 'com.google.api.grpc:proto-google-common-protos:2.9.6'
   api 'com.google.api.grpc:proto-google-iam-v1:1.6.2'
diff --git a/x-pack/plugin/ml/build.gradle b/x-pack/plugin/ml/build.gradle
index c13d178c3b2..83c4075e0f7 100644
--- a/x-pack/plugin/ml/build.gradle
+++ b/x-pack/plugin/ml/build.gradle
@@ -1,5 +1,3 @@
-import org.elasticsearch.gradle.VersionProperties
-
 apply plugin: 'elasticsearch.internal-es-plugin'
 apply plugin: 'elasticsearch.internal-cluster-test'
 apply plugin: 'elasticsearch.internal-test-artifact'
@@ -13,8 +11,6 @@ esplugin {
   extendedPlugins = ['x-pack-autoscaling', 'lang-painless']
 }
 
-def localRepo = providers.systemProperty('build.ml_cpp.repo').orNull
-
 if (useDra == false) {
   repositories {
     exclusiveContent {
@@ -24,26 +20,13 @@ if (useDra == false) {
       forRepository {
         ivy {
           name "ml-cpp"
+          url providers.systemProperty('build.ml_cpp.repo').orElse('https://prelert-artifacts.s3.amazonaws.com').get()
           metadataSources {
             // no repository metadata, look directly for the artifact
             artifact()
           }
-          if (localRepo) {
-            url localRepo
-            patternLayout {
-              artifact "maven/[orgPath]/[module]/[revision]/[module]-[revision](-[classifier]).[ext]"
-            }
-          } else {
-            url "https://artifacts-snapshot.elastic.co/"
-            patternLayout {
-              if (VersionProperties.isElasticsearchSnapshot()) {
-                artifact '/ml-cpp/[revision]/downloads/ml-cpp/[module]-[revision](-[classifier]).[ext]'
-              } else {
-                // When building locally we always use snapshot artifacts even if passing `-Dbuild.snapshot=false`.
-                // Release builds are always done with a local repo.
-                artifact '/ml-cpp/[revision]-SNAPSHOT/downloads/ml-cpp/[module]-[revision]-SNAPSHOT(-[classifier]).[ext]'
-              }
-            }
+          patternLayout {
+            artifact "maven/org/elasticsearch/ml/ml-cpp/[revision]/ml-cpp-[revision].[ext]"
           }
         }
       }
diff --git a/x-pack/plugin/security/build.gradle b/x-pack/plugin/security/build.gradle
index 431a64dd98a..16ad9545036 100644
--- a/x-pack/plugin/security/build.gradle
+++ b/x-pack/plugin/security/build.gradle
@@ -53,7 +53,7 @@ dependencies {
   api "org.opensaml:opensaml-storage-api:3.4.5"
   api "org.opensaml:opensaml-storage-impl:3.4.5"
   api "net.shibboleth.utilities:java-support:7.5.1"
-  api "org.apache.santuario:xmlsec:2.1.4"
+  api "org.apache.santuario:xmlsec:2.1.8"
   api "io.dropwizard.metrics:metrics-core:3.2.2"
   api ("org.cryptacular:cryptacular:1.2.4") {
     exclude group: 'org.bouncycastle'
@@ -74,7 +74,7 @@ dependencies {
   api "com.sun.mail:jakarta.mail:1.6.3"
   api "net.jcip:jcip-annotations:1.0"
   api "net.minidev:json-smart:2.4.10"
-  api "net.minidev:accessors-smart:2.4.2"
+  api "net.minidev:accessors-smart:2.4.8"
   api "org.ow2.asm:asm:9.1"
 
   testImplementation "org.elasticsearch:mocksocket:${versions.mocksocket}"
diff --git a/x-pack/plugin/vector-tile/build.gradle b/x-pack/plugin/vector-tile/build.gradle
index c4833440c24..6fca9f6bee9 100644
--- a/x-pack/plugin/vector-tile/build.gradle
+++ b/x-pack/plugin/vector-tile/build.gradle
@@ -29,11 +29,11 @@ dependencies {
   testImplementation(testArtifact(project(xpackModule('core'))))
   compileOnly "org.locationtech.jts:jts-core:${versions.jts}"
   api "com.wdtinc:mapbox-vector-tile:3.1.0"
-  api "com.google.protobuf:protobuf-java:3.16.3"
+  api "com.google.protobuf:protobuf-java:3.21.9"
   runtimeOnly("org.slf4j:slf4j-api:${versions.slf4j}")
   runtimeOnly("org.apache.logging.log4j:log4j-slf4j-impl:${versions.log4j}")
   javaRestTestImplementation("com.wdtinc:mapbox-vector-tile:3.1.0")
-  javaRestTestImplementation("com.google.protobuf:protobuf-java:3.16.3")
+  javaRestTestImplementation("com.google.protobuf:protobuf-java:3.21.9")
 }
 
 testClusters.configureEach {
diff --git a/x-pack/snapshot-tool/build.gradle b/x-pack/snapshot-tool/build.gradle
index abca20b9026..a888e90ad45 100644
--- a/x-pack/snapshot-tool/build.gradle
+++ b/x-pack/snapshot-tool/build.gradle
@@ -42,8 +42,8 @@ dependencies {
   api 'com.google.api:api-common:2.3.1'
   api 'com.google.api:gax:2.20.1'
   api 'org.threeten:threetenbp:1.6.5'
-  api 'com.google.protobuf:protobuf-java-util:3.17.3'
-  api 'com.google.protobuf:protobuf-java:3.21.7'
+  api 'com.google.protobuf:protobuf-java-util:3.21.9'
+  api 'com.google.protobuf:protobuf-java:3.21.9'
   api 'com.google.code.gson:gson:2.10'
   api 'com.google.api.grpc:proto-google-common-protos:2.9.6'
   api 'com.google.api.grpc:proto-google-iam-v1:1.6.2'
-- 
2.43.0

